internal/handlers/jsonb1/msg_count.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 FerretDB Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package jsonb1
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v4"
+
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/pg"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
+	"github.com/FerretDB/FerretDB/internal/wire"
+)
+
+// MsgCount counts the number of documents that matches the query filter.
+func (s *storage) MsgCount(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err := common.Unimplemented(document, common.UnimplementedFieldsFind...); err != nil {
+		return nil, err
+	}
+	common.Ignored(document, s.l, common.IgnoredFieldsFind...)
+
+	var filter *types.Document
+	var sql, collection string
+	var args []any
+	var placeholder pg.Placeholder
+
+	m := document.Map()
+	collection = m["count"].(string)
+	db := m["$db"].(string)
+
+	collection = m["count"].(string)
+	sql = fmt.Sprintf(`SELECT COUNT(*) FROM %s`, pgx.Identifier{db, collection}.Sanitize())
+	filter, _ = m["query"].(*types.Document)
+
+	whereSQL, whereArgs, err := where(filter, &placeholder)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	args = append(args, whereArgs...)
+	sql += whereSQL
+
+	tailSQL, tailArgs, err := common.Tail(" _jsonb->", m, &placeholder)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	args = append(args, tailArgs...)
+	sql += tailSQL
+
+	rows, err := s.pgPool.Query(ctx, sql, args...)
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+	defer rows.Close()
+
+	var reply wire.OpMsg
+	var count int32
+	for rows.Next() {
+		err := rows.Scan(&count)
+		if err != nil {
+			return nil, lazyerrors.Error(err)
+		}
+	}
+	// in psql, the SELECT * FROM table limit `x` ignores the value of the limit,
```

Comment:
psql should not ignore `LIMIT` in `SELECT * FROM table LIMIT x`. WDYM?

Reply:
it's strange, is should have been changed to 	`return common.QueryCount(s.pgPool, ctx, sql, args)' as I have in my local code..

Reply:
@AlekSi no changes here regarding previous code, it worked so for when all was in a single msg_find_or_count.go.

Reply:
removed it.